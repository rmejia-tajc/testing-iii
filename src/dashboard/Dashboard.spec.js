// Test away!

import React from "react";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Dashboard from "./Dashboard.js";

describe("<Dashboard /> Component", () => {

  it("renders without crashing", () => {
    const sanityCheck = render(<Dashboard />);
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<Dashboard />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it.only("should default to (lock gate)[unlocked] and (close gate)[open]", () => {
    const { getByTestId } = render(<Dashboard />);

    const unlockedLocked = getByTestId(/unlockedLocked/i);
    const openClosed = getByTestId(/openClosed/i);
    const btnLockUnlock = getByTestId(/lockUnlockGate/i);
    const btnCloseOpen = getByTestId(/closeOpenGate/i);

    expect(unlockedLocked).toBeInTheDocument();
    expect(openClosed).toBeInTheDocument();
    expect(btnLockUnlock).toBeInTheDocument();
    expect(btnCloseOpen).toBeInTheDocument();
  });

  it('the "close gate" button should change to "open gate" when clicked', () => {
    const { getByTestId, queryByText } = render(<Dashboard />);

    expect(queryByText(/close gate/i)).toBeTruthy();
    expect(queryByText(/open gate/i)).toBeFalsy();

    const btnCloseOpen = getByTestId(/closeOpenGate/i);
    fireEvent.click(btnCloseOpen);

    expect(queryByText(/close gate/i)).toBeFalsy();
    expect(queryByText(/open gate/i)).toBeTruthy();
  });

});
