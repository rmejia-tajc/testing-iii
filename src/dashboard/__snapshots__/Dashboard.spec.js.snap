// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Component should match snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="unlockedLocked"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="openClosed"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lockUnlockGate"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="closeOpenGate"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
